### DISCUSSION TITLE
Question about combining `autoGainControl` (from `mediaDevices.getUserMedia`) and `DynamicsCompressorNode`

### DISCUSSION THREAD

**Discussion by grazder** [WebAudio/web-audio-api#2597] [General]
Hello! I need help understanding the gain parameters for `autoGainControl` (from `mediaDevices.getUserMedia`) and `DynamicsCompressorNode`.

I have a pipeline of 
```
Source -> `autoGainControl: True` -> some other nodes -> `DynamicsCompressorNode` -> Destination
```
`autoGainControl` sets the volume of the source audio to a certain volume range. Then I run several nodes that change this volume range (for example, I make the audio quieter). After that, I use `DynamicsCompressorNode` to compress the audio AND to boost the volume of the audio, since auto makeup gain is built into `DynamicsCompressorNode`.

And with this pipeline I want that at the destination after the full pipeline (`Source -> autoGainControl: True -> some other nodes -> DynamicsCompressorNode`) I have the same volume as after autoGain (`Source -> autoGainControl: True -> Destination`).

Can I do this? Can you help me understand what `DynamicsCompressorNode` parameters I should use for this? And with what limitations?

**Comment by grazder** [comment]
https://chromium.googlesource.com/external/webrtc/+/lkgr/modules/audio_processing/agc/agc.cc#24

If `autoGainControl` forces audio to -18dB, so I need to set `DynamicsCompressorNode.threshold.value` = -18? Is this the solution or am I missing something?

**Comment by grazder** [comment]
@padenot

**Comment by padenot** [comment]
The only way to do this is to measure what you have at the output of your pipeline and adjust. `autoGainControl` isn't specified precisely and it will differ frequently from one situation to the next (browsers, OS, browser version, OS version, etc.).

**Reply by grazder** [reply]
Thank you!
